app/models/consent_form.rb

```diff
@@ -23,34 +23,57 @@
 #
 
 class ConsentForm < ApplicationRecord
-  attr_accessor :is_this_their_school
+  cattr_accessor :form_steps do
+    %i[name date_of_birth school]
+  end
+
+  attr_accessor :form_step, :is_this_their_school
 
   audited
 
   belongs_to :session
 
-  validates :first_name, presence: true, on: :name
-  validates :last_name, presence: true, on: :name
-  validates :use_common_name, inclusion: { in: [true, false] }, on: :name
-  validates :common_name, presence: true, on: :name, if: :use_common_name?
-
-  validates :date_of_birth,
-            presence: true,
-            comparison: {
-              less_than: Time.zone.today,
-              greater_than_or_equal_to: 22.years.ago.to_date,
-              less_than_or_equal_to: 3.years.ago.to_date
-            },
-            on: :date_of_birth
-
-  validates :is_this_their_school,
-            presence: true,
-            inclusion: {
-              in: %w[yes no]
-            },
-            on: :school
+  with_options if: -> { required_for_step?(:name) } do
+    validates :first_name, presence: true
+    validates :last_name, presence: true
+    validates :use_common_name, inclusion: { in: [true, false] }
+    validates :common_name, presence: true, if: :use_common_name?
+  end
+
+  with_options if: -> { required_for_step?(:date_of_birth) } do
+    validates :date_of_birth,
+              presence: true,
+              comparison: {
+                less_than: Time.zone.today,
+                greater_than_or_equal_to: 22.years.ago.to_date,
+                less_than_or_equal_to: 3.years.ago.to_date
+              }
+  end
+
+  with_options if: -> { required_for_step?(:school, exact: true) } do
+    validates :is_this_their_school,
+              presence: true,
+              inclusion: {
+                in: %w[yes no]
+              }
+  end
 
   def full_name
     [first_name, last_name].join(" ")
   end
+
+  private
+
+  def required_for_step?(step, exact: false)
+    # Exact means that the form_step must match the step
+    return false if exact && form_step != step
+
+    # All fields are required if no form_step is set
+    return true if form_step.nil?
+
+    # Otherwise, all fields from previous and current steps are required
+    return true if form_steps.index(step) <= form_steps.index(form_step)
+
+    false
+  end
```

Comment:
I found this a little difficult to read, sorry to say, although maybe that was just the fog left-over from the meeting earlier. 

1. I think `required_for_step?(step, ...)` gives the false impression that `step` is what's being tested, when in fact it's the form step that's being tested, hence the name change. 
2. With the return statements written the way the are, I was having a hard time understanding what `exact` did, er ... exactly. Testing it separately makes it clearer, although I'm not sure if Rubocop might not try to golf that down for us.

```suggestion
  def form_step_matches?(step, exact: false)
    # All fields are required if no form_step is set
    return true if form_step.nil?

    if exact
      # Exact means that the form_step must match the step
      return form_step == step
    else
      # Otherwise, all fields from previous and current steps are required
      return form_steps.index(step) <= form_steps.index(form_step)
  end
```